php/alsearch.php

```diff
@@ -164,63 +171,62 @@
     exit;
 }
 
-$sql = "SELECT * FROM airlines WHERE ";
-$params = [];
+$filters = [];
+$filterParams = [];
 
 // Build filter
 if ($name) {
-    $sql .= " (name LIKE ? OR alias LIKE ?) AND";
-    $params[] = $name . '%';
-    $params[] = $name . '%';
+    $filters[] = "(name LIKE ? OR alias LIKE ?)";
+    $filterParams[] = $name . '%';
+    $filterParams[] = $name . '%';
 }
 if ($alias) {
-    $sql .= " (name LIKE ? OR alias LIKE ?) AND";
-    $params[] = $alias . '%';
-    $params[] = $alias . '%';
+    $filters[] = "(name LIKE ? OR alias LIKE ?)";
+    $filterParams[] = $alias . '%';
+    $filterParams[] = $alias . '%';
 }
 if ($callsign) {
-    $sql .= " callsign LIKE ? AND";
-    $params[] = $callsign . '%';
+    $filters[] = "callsign LIKE ?";
+    $filterParams[] = $callsign . '%';
 }
 
 if ($iata) {
-    $sql .= " iata=? AND";
-    $params[] = $iata;
+    $filters[] = "iata = ?";
+    $filterParams[] = $iata;
 }
 if ($icao) {
-    $sql .= " icao=? AND";
-    $params[] = $icao;
+    $filters[] = "icao = ?";
+    $filterParams[] = $icao;
 }
 if ($country != "ALL" && $country) {
-    $sql .= " country=? AND";
-    $params[] = $country;
+    $filters[] = "country = ?";
+    $filterParams[] = $country;
 }
 if ($mode) {
-    $sql .= " mode=? AND";
-    $params[] = $mode;
+    $filters[] = "mode = ?";
+    $filterParams[] = $mode;
 }
 if ($active != "") {
-    $sql .= " active=? AND";
-    $params[] = $active;
+    $filters[] = "active = ?";
+    $filterParams[] = $active;
 }
 
-if ($mode != "F" || $iatafilter == "false") {
-    $sql .= " 1=1"; // dummy
-} else {
-    $sql .= " iata != '' AND iata != 'N/A'";
+if ($mode == "F" && $iatafilter != "false") {
+    $filters[] = "iata != ''";
+    $filters[] = "iata != 'N/A'";
```

Comment:
Perhaps a good fit for `NOT IN`?

```suggestion
    $filters[] = "iata NOT IN ('', 'N/A')";
```